client/src/pages/board/KanbanBoard.jsx

```diff
@@ -0,0 +1,297 @@
+          padding: '0 30px 0 0',
+          left: '-30px',
+          minHeight: '500px',
+          position: 'fixed',
+          textDecoration: 'none',
+          height: 'inherit',
+          display: 'flex',
+          alignItems: 'center',
+          '& button': {
+            marginRight: '10px',
+            backgroundColor: 'lightgrey',
+            padding: '0px',
+            color: 'grey',
+          },
+          '&:hover': {
+            padding: '0 0 0 20px',
+            left: 0,
+            position: 'relative',
+            width: 'fit-content',
+            backgroundColor: 'lightgrey',
+            transition: '0.05s',
+          },
+        },
+      },
+    },
+    '&#rightNav': {
+      display: 'flex',
+      justifyContent: 'flex-end',
+      '& .addColumnContent': {
+        minHeight: '500px',
+        '& a': {
+          position: 'fixed',
+          transition: '0.1s',
+          textDecoration: 'none',
+          height: 'inherit',
+          display: 'flex',
+          alignItems: 'center',
+          width: '2%',
+          padding: '0 0 0 30px',
+          right: '-30px',
+          minHeight: '500px',
+          '& button': {
+            marginLeft: '10px',
+            backgroundColor: 'lightgrey',
+            color: 'grey',
+            padding: '0px',
+          },
+          '&:hover': {
+            position: 'relative',
+            width: 'fit-content',
+            backgroundColor: 'lightgrey',
+            right: 0,
+            padding: '0 10px 0 0',
+          },
+        },
+      },
+    },
+  },
+}));
+
+// performance optimization. prevents re-render when components are dragged all over w/memo
+const InnerList = memo((props) => {
+  const { column, taskMap, index, setUpdate = { setUpdate } } = props;
+  const tasks = column.taskIds.map((taskId) => taskMap[taskId]);
+  return <Column column={column} tasks={tasks} index={index} setUpdate={setUpdate} />;
+});
+
+export default function KanbanBoard() {
+  const [data, setData] = useState({
+    id: '',
+    tasks: {},
+    columns: {},
+    columnOrder: []
+  });
+
+  const [update, setUpdate] = useState(true)
+
+  const convertAPIData = async () => {
+    const boardData = await getBoard()
+    
+    const data = await boardData.data
+    
+    const newData = {
+      id: data._id,
+      tasks: {},
+      columns: {},
+      columnOrder: [],
+    }
+    await data.columns.map(column => {
+      newData.columns[column._id] = {
+        id: column._id,
+        title: column.name,
+        taskIds: column.cards.map(card => card._id)
+      }
+      column.cards.map(card => {
+        newData.tasks[card._id] = {
+          id: card._id,
+          content: card.name
+        }
+      })
+      newData.columnOrder.push(column._id)
+    })
+    setData(newData)
+  }
+
+  useEffect(() => {
+    if (update) {
+      convertAPIData()
+      setUpdate(false)
+    }
+  }, [update])
+
+  const classes = useStyles();
+  const onDragEnd = async (result) => {
```

Comment:
I'll open a new issue for this and leave it for another time. I'm thinking when we get closer to the end of adding all of our features. Currently it will be easier to follow the onDragEnd documentation side-by-side without refactoring if any issues do arise.